api_app/services/aad_authentication.py

```diff
@@ -282,27 +282,44 @@ def _get_app_auth_info(self, client_id: str) -> dict:
 
         return authInfo
 
-    def _get_role_assignment_graph_data_for_user(self, user_id: str) -> dict:
+    def _ms_graph_query(self, url: str, http_method: str, json=None) -> dict:
         msgraph_token = self._get_msgraph_token()
+        auth_headers = self._get_auth_header(msgraph_token)
+        graph_data = {}
+        while True:
+            if not url:
+                break
+            logging.debug(f"Making request to: {url}")
+            if json:
+                response = requests.request(method=http_method, url=url, json=json, headers=auth_headers)
+            else:
+                response = requests.request(method=http_method, url=url, headers=auth_headers)
+            url = ""
+            if response.status_code == 200:
+                json_response = response.json()
+                graph_data = merge_dict(graph_data, json_response)
+                if '@odata.nextLink' in json_response:
+                    url = json_response['@odata.nextLink']
+            else:
+                logging.error(f"MS Graph query to: {url} failed with status code {response.status_code}")
```

Comment:
Is it worth including the response body? (I think it did this previously to capture the error message)

Reply:
done.